keps/sig-node/3695-pod-resources-for-dra/README.md

```diff
@@ -0,0 +1,372 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+We propose an enhancement to the pod resource api to include resources allocated by [Dynamic Resource Allocation (DRA)](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/3063-dynamic-resource-allocation). We also propose to extend the API to be more friendly for consumption by CNI meta-plugins.
+This KEP extends [2043-pod-resource-concrete-assigments](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2043-pod-resource-concrete-assigments) and [2403-pod-resources-allocatable-resources](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2403-pod-resources-allocatable-resources).
+
+## Motivation
+
+The PodResources API is currently being used by CNI meta-plugins like [multus](https://github.com/k8snetworkplumbingwg/multus-cni) and [DANM](https://github.com/nokia/danm) to add resources allocated by device plugins as CNI arguments. One of the primary motivations for this KEP is to extend the PodResources API to allow these CNI plugins to reference resources allocated by DRA as well. Additional extensions to API will also make it easier for CNI meta-plugins to access resources allocated to a specific pod, rather than having to filter through resources for all pods on the node. Another use-case is to allow node monitoring agents to access information about allocated resources.
+
+### Goals
+
+Enabling the DRA feature to work with CNIs that require complex network devices such as RDMA. DRA resource drivers will allocate the resources, and the meta-plugin will read the allocated [CDI Device](https://github.com/container-orchestrated-devices/container-device-interface) using the Pod resource API. The meta-plugin will then inject the device-id as CNI arguments and invoke other CNIs (just as it does for devices allocated by the device plugin today).
+
```

Comment:
We need to chat with sig-network about that in more detail. this approach for me looks like a hack rather than long term solution.

What we had in mind for DRA for network, is to enable DRA claim to be consumable as Pod level resource, like all network interfaces are. This way, the DRA driver can expose needed bits to Pod metadata that would be available in CNI, as well, DRA can be used as generic approach to request particular Network interface(s) based on specified properties, without relaying on some CNI side multiplexors.

Reply:
This is short term solution until multi-network will be supported in k8s.

Do you have design proposal or flow diagram of DRA for network? It is not clear to me how the pod metadata would be available in CNI. How it will work with multi-network? how do you link between the resource claim and the network?

why we will want to use DRA as generic approach to request particular Network interface?
It is mixing resources preparation with multi-network how to request network (which there is discussion https://github.com/kubernetes/enhancements/pull/3700)

Reply:
I agree that we should consult with sig-network on whether this is the right approach long-term for CNI.

With that being said, I do believe this extension to the PodResources API has merit without the implications for CNI (e.g. for metrics).

@moshe010 can you turn this section into a bulleted list and add a goal above this one that outlines the metrics use case.